scripts/generator/graphql_generator/csharp/CartState.cs.erb

```diff
@@ -300,7 +300,9 @@ namespace <%= namespace %> {
                 List<CheckoutLineItem> lineItems = (List<CheckoutLineItem>) checkout.lineItems();
 
                 LineItems.UpdateLineItemsFromCheckoutLineItems(lineItems);
+            #pragma warning disable 0168
             } catch(NoQueryException exception) {}
```

Comment:
This is because NoQueryException is defined in a generated file? can you explain why this warning occurs and why we need to disable it rather than provide a fix?